[CI] WildcardFieldMapperTests testSearchResultsVersusKeywordField failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/tpwunmmeicxm6/tests/:x-pack:plugin:wildcard:test/org.elasticsearch.xpack.wildcard.mapper.WildcardFieldMapperTests/testSearchResultsVersusKeywordField

**Reproduction line:**
`./gradlew ':x-pack:plugin:wildcard:test' --tests "org.elasticsearch.xpack.wildcard.mapper.WildcardFieldMapperTests.testSearchResultsVersusKeywordField" -Dtests.seed=AA2393E9F981C740 -Dtests.locale=es-UY -Dtests.timezone=Australia/Brisbane -Druntime.java=16`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.wildcard.mapper.WildcardFieldMapperTests&tests.test=testSearchResultsVersusKeywordField

**Failure excerpt:**
```
java.lang.AssertionError: (No message provided)

  at __randomizedtesting.SeedInfo.seed([AA2393E9F981C740:734BEFC972D0E077]:0)
  at org.elasticsearch.xpack.wildcard.mapper.BinaryDvConfirmedAutomatonQuery$1$1.matches(BinaryDvConfirmedAutomatonQuery.java:89)
  at org.apache.lucene.search.ConstantScoreScorer$1.matches(ConstantScoreScorer.java:103)
  at org.apache.lucene.search.Weight$DefaultBulkScorer.scoreRange(Weight.java:281)
  at org.apache.lucene.search.Weight$DefaultBulkScorer.score(Weight.java:254)
  at org.apache.lucene.search.AssertingBulkScorer.score(AssertingBulkScorer.java:97)
  at org.apache.lucene.search.AssertingBulkScorer.score(AssertingBulkScorer.java:63)
  at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:756)
  at org.apache.lucene.search.AssertingIndexSearcher.search(AssertingIndexSearcher.java:73)
  at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:679)
  at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:673)
  at org.apache.lucene.search.IndexSearcher.searchAfter(IndexSearcher.java:653)
  at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:570)
  at org.elasticsearch.xpack.wildcard.mapper.WildcardFieldMapperTests.testSearchResultsVersusKeywordField(WildcardFieldMapperTests.java:412)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:567)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:831)

```

--- a/x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/BinaryDvConfirmedAutomatonQuery.java
+++ b/x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/BinaryDvConfirmedAutomatonQuery.java
@@ -15,7 +15,6 @@ import org.apache.lucene.search.ConstantScoreScorer;
 import org.apache.lucene.search.ConstantScoreWeight;
 import org.apache.lucene.search.DocIdSetIterator;
 import org.apache.lucene.search.IndexSearcher;
-import org.apache.lucene.search.MatchAllDocsQuery;
 import org.apache.lucene.search.Query;
 import org.apache.lucene.search.ScoreMode;
 import org.apache.lucene.search.Scorer;
@@ -84,8 +83,7 @@ public class BinaryDvConfirmedAutomatonQuery extends Query {
                     public boolean matches() throws IOException {
                         if (values.advanceExact(approxDisi.docID()) == false)
                         {
-                            // Bug if we have an indexed value (i.e an approxQuery) but no doc value.
-                            assert approxQuery instanceof MatchAllDocsQuery;
+                            // Can happen when approxQuery resolves to some form of MatchAllDocs expression
                             return false;
                         }
                         BytesRef arrayOfValues = values.binaryValue();

--- a/x-pack/plugin/wildcard/src/test/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapperTests.java
+++ b/x-pack/plugin/wildcard/src/test/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapperTests.java
@@ -311,7 +311,6 @@ public class WildcardFieldMapperTests extends MapperTestCase {
         assertThat(td.totalHits.value, equalTo(count));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/78949")
     public void testSearchResultsVersusKeywordField() throws IOException {
         Directory dir = newDirectory();
         IndexWriterConfig iwc = newIndexWriterConfig(WildcardFieldMapper.WILDCARD_ANALYZER_7_10);

